src/ui/public/agg_types/metrics/lib/parent_pipeline_agg_helper.js

```diff
@@ -0,0 +1,61 @@
+import metricAggTemplate from 'ui/agg_types/controls/sub_agg.html';
+import _ from 'lodash';
+import VisAggConfigProvider from 'ui/vis/agg_config';
+import VisSchemasProvider from 'ui/vis/schemas';
+import { parentPipelineAggController } from './lib/parent_pipeline_agg_controller';
+import { parentPipelineAggWritter } from './lib/parent_pipeline_agg_writter';
+
+const ParentPipelineAggHelperProvider = function (Private) {
```

Comment:
Just to confirm I understand the term `parentPipeline`, it  s when in the Y-Axis you have a sub aggregation, and right now is only derivatives, but you split this logic out because it will be relevant to the cumulative sum you mentioned that you are also adding?